jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -767,6 +767,209 @@ ecma_builtin_typedarray_prototype_set (ecma_value_t this_arg, /**< this argument
   return ret_val;
 } /* ecma_builtin_typedarray_prototype_set */
 
+/**
+ * TypedArray.prototype's 'toString' single element operation routine based
+ * on the Array.prototype's 'toString' single element operation routine
+ *
+ * See also:
+ *          ECMA-262 v5.1, 15.4.4.2
+ *
+ * @return ecma_value_t value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_op_typedarray_get_to_string_at_index (ecma_object_t *obj_p, /**< this object */
+                                           uint32_t index) /**< array index */
+{
+  ecma_value_t ret_value = ECMA_VALUE_EMPTY;
+  ecma_string_t *index_string_p = ecma_new_ecma_string_from_uint32 (index);
+
+  ECMA_TRY_CATCH (index_value,
```

Comment:
We've started the elimination of `ECMA_TRY_CATCH`, so it would be better to avoid the usage of it.

Reply:
I'll look up the whole file for it then, is there a doc or something about the usage of the other method?

Reply:
There is no documentation, but it easy to avoid the usage of it. Check the PR of @galpeter #2246. You must check the returned `ecma_value_t` with `ECMA_IS_VALUE_ERROR` and return if it is an error. Don't forget to free everything before you return. E.g:

```c
static ecma_value_t
ecma_op_typedarray_get_to_string_at_index (ecma_object_t *obj_p, /**< this object */
                                           uint32_t index) /**< array index */
{
  ecma_string_t *index_string_p = ecma_new_ecma_string_from_uint32 (index);
  ecma_value_t index_value = ecma_op_object_get (obj_p, index_string_p);
  ecma_deref_ecma_string (index_string_p);

  if (ECMA_IS_VALUE_ERROR (index_value))
  {
    return index_value;
  }

  if (ecma_is_value_undefined (index_value)
     || ecma_is_value_null (index_value))
  {
    ecma_free_value (index_value);
    return ecma_make_magic_string_value (LIT_MAGIC_STRING__EMPTY);
  }

  ecma_free_value (index_value);
  return ecma_op_to_string (index_value);
} /* ecma_op_typedarray_get_to_string_at_index */
```